platform/features/layout/src/FixedController.js

```diff
@@ -311,26 +288,228 @@ define(
+         * @param {object} datum The telemetry datum containing the values to print
+         */
+        FixedController.prototype.updateView = function (telemetryObject, datum) {
+            var metadata = this.openmct.telemetry.getMetadata(telemetryObject);
+            var rangeMetadata = metadata.valuesForHints(['range'])[0];
+            var rangeKey = rangeMetadata.source || rangeMetadata.key;
+            var valueMetadata = metadata.value(rangeKey);
+            var limitEvaluator = this.openmct.telemetry.limitEvaluator(telemetryObject);
+            var formatter = this.openmct.telemetry.getValueFormatter(valueMetadata);
+            var value = datum[valueMetadata.key];
+            var alarm = limitEvaluator && limitEvaluator.evaluate(datum, rangeKey);
+
+            this.setDisplayedValue(
+                telemetryObject,
+                formatter.format(value),
+                alarm && alarm.cssClass
+            );
+            this.digest();
+        };
+
+        /**
+         * Returns a new or cached telemetry collection associated with a given
+         * object.
+         * @private
+         * @param {object} object A domain object
+         * @returns {TelemetryCollection} the collection of telemetry for the given object
+         */
+        FixedController.prototype.getCollectionForObject = function (object) {
+            var telemetryCollection = this.telemetryCollections.get(object);
+            var timeSystem;
+            var domainKey;
+            var metadata;
+            var domainMetadata;
+            var self = this;
+
+            if (telemetryCollection === undefined) {
+                timeSystem = this.openmct.conductor.timeSystem();
+                metadata = this.openmct.telemetry.getMetadata(object);
+                domainMetadata = metadata.valuesForHints(['domain']).filter(function (value) {
+                    return value.key === timeSystem.metadata.key;
+                })[0];
+
+                if (domainMetadata) {
+                    domainKey = domainMetadata.source || domainMetadata.key;
+                } else {
+                    domainKey = timeSystem.metadata.key;
+                }
+
+                telemetryCollection = new TelemetryCollection();
+                telemetryCollection.bounds(this.openmct.conductor.bounds());
+                telemetryCollection.sort(domainKey);
+                telemetryCollection.on("added", function (telemetryAdded) {
+                    var datum = telemetryAdded[telemetryAdded.length - 1];
+                    if (datum !== undefined) {
+                        self.updateView(object, datum);
+                    }
+
+                    // Optimization, clear the cached telemetry to free
+                    // up memory. We're only interested in the last value <
+                    // bounds.end
+                    telemetryCollection.telemetry = [];
```

Comment:
You make a very good point. The `TelemetryCollection` actually does define a `.clear()` method, but as well as clearing the telemetry array, it also clears its internal buffer, which I don't want to do in this case. The buffer keeps telemetry that is ahead of the current time bounds, which may soon fall within bounds. This happens because the tick source and telemetry source clocks are not necessarily synchronized. 

The idea with the `TelemetryCollection` is that it is backed by a vanilla array to allow fast array operations, with the underlying array being exposed directly to allow array operations. So the intention is for the `.telemetry` attribute to be part of `TelemetryCollection`s public API. 

Perhaps I could make this more explicit with a getter/setter function?